build-system/tasks/visual-diff.js

```diff
@@ -509,6 +509,12 @@ async function visualDiffPuppeteer() {
     log('fatal', 'Could not find', colors.cyan('PERCY_PROJECT'), 'and',
         colors.cyan('PERCY_TOKEN'), 'environment variables');
   }
+
+  if (!AMP_RUNTIME_TARGET_FILES.every(fs.existsSync)) {
+    log('fatal', 'Must run', colors.cyan('gulp build'), 'before running',
+        colors.cyan('gulp visual-diff'));
+  }
```

Comment:
You can directly specify prerequisite build tasks as a requirement of a `gulp` task...

https://github.com/ampproject/amphtml/blob/7635b8e7f6b9cc74e975b32868ce86b541bd43db/build-system/tasks/runtime-test.js#L641

... and not run them when `--nobuild` is passed in...

https://github.com/ampproject/amphtml/blob/7635b8e7f6b9cc74e975b32868ce86b541bd43db/build-system/tasks/runtime-test.js#L40-L41

... and educate the user that `--nobuild` is an option if the task contains `visual-diff`...

https://github.com/ampproject/amphtml/blob/7635b8e7f6b9cc74e975b32868ce86b541bd43db/gulpfile.js#L720-L729

This way, instead of failing when a build is not found, or running visual tests against a stale build, you can build the runtime as a pre-requisite of `gulp visual-diff`, and skip it if `--nobuild` is passed in.